alternator/executor.cc

```diff
@@ -1636,19 +1649,56 @@ future<executor::request_return_type> rmw_operation::execute(service::storage_pr
     });
 }
 
+bytes query_cache_key(const std::string_view& query, char query_type) {
+   auto b = md5_hasher::calculate(query);
+   b[0] = query_type; // this avoids key collision for different expression types
```

Comment:
Is expression parsing different for different types? Or are the types only important in how the expressions are used?

I agree there's no need to calculate the md5 hash. It's used in CQL because that's how the protocol is defined (and in turn, this allows the client to avoid sending the entire query string each time), but here we don't have a protocol with md5, and we also happen to have the query string anyway.

Reply:
Yes, there are exactly three types of expressions: UpdateExpression, ProjectionExpression and ConditionExpression. Each one has different syntax to parse and different output type after the parsing.